hoot-core/src/main/cpp/hoot/core/geometry/GeometryUtils.cpp

```diff
@@ -228,20 +245,143 @@ Envelope GeometryUtils::envelopeFromConfigString(const QString& boundsStr)
+  std::shared_ptr<Polygon> poly(
+    GeometryFactory::getDefaultInstance()->createPolygon(outer, holes));
+  LOG_VAR(poly->isValid());
+  return poly;
+}
+
+std::shared_ptr<Polygon> GeometryUtils::polygonFromString(const QString& str)
+{
+  // format: x1,y1;x2,y2;x3,y3...
+
+  // You could make the argument to use something more standardized here for the format, like WKT.
+  // This format is easier to read and parse. If we decide to support geometries other than polys,
+  // then we may need to use a different format.
+
+  LOG_VART(str);
+  if (str.trimmed().isEmpty())
+  {
+    return std::shared_ptr<Polygon>();
+  }
+
+  QStringList coords = str.split(";");
+  LOG_VART(coords.size());
+  if (coords.size() < 3)
+  {
+    throw IllegalArgumentException("Polygon string must have at least three points: " + str);
+  }
+
+  LOG_VART(coords.at(0) != coords.at(coords.size() - 1));
+  if (coords.at(0) != coords.at(coords.size() - 1))
+  {
+    coords.append(coords.at(0));
+  }
+  LOG_VART(coords);
+
+  CoordinateSequence* coordSeq =
+    GeometryFactory::getDefaultInstance()->getCoordinateSequenceFactory()->create(coords.size(), 2);
+  for (int i = 0; i < coords.size(); i++)
+  {
+    const QString coordStr = coords.at(i);
+    const QStringList coordStrs = coordStr.split(",");
+    if (coordStrs.size() != 2)
+    {
+      throw IllegalArgumentException("Invalid polygon coordinate string: " + coordStr);
+    }
+    bool ok = false;
+    const double x = coordStrs.at(0).toDouble(&ok);
+    if (!ok)
+    {
+      throw IllegalArgumentException("Invalid polygon x coordinate value: " + coordStrs.at(0));
+    }
+    const double y = coordStrs.at(1).toDouble(&ok);
+    if (!ok)
+    {
+      throw IllegalArgumentException("Invalid polygon y coordinate value: " + coordStrs.at(1));
+    }
+    coordSeq->setAt(geos::geom::Coordinate(x, y), i);
+  }
+  LOG_VART(coordSeq->size());
+
+  // an empty set of holes
+  vector<Geometry*>* holes = new vector<Geometry*>();
+  // create the outer line
+  LinearRing* outer = GeometryFactory::getDefaultInstance()->createLinearRing(coordSeq);
+  return std::shared_ptr<Polygon>(
+    GeometryFactory::getDefaultInstance()->createPolygon(outer, holes));
+}
+
+QString GeometryUtils::polygonStringToEnvelopeString(const QString& str)
+{
+  return envelopeToString(*(polygonFromString(str)->getEnvelopeInternal()));
+}
+
+QString GeometryUtils::polygonToString(const std::shared_ptr<Polygon>& poly)
+{
+  const int precision = ConfigOptions().getWriterPrecision();
+  geos::geom::CoordinateSequence* coords = poly->getCoordinates();
+  QString str;
+  for (size_t i = 0; i < coords->getSize(); i++)
+  {
+    const geos::geom::Coordinate& coord = coords->getAt(i);
+    str +=
+      QString::number(coord.x, 'g', precision) + "," + QString::number(coord.y, 'g', precision) +
+      ";";
+  }
+  str.chop(1);
+  return str;
+}
+
+std::shared_ptr<geos::geom::Polygon> GeometryUtils::boundsFromString(const QString& str)
+{
```

Comment:
Coming from a config string, only supporting `Envelope` and `Polygon` for now, as that's all that is needed. Can be changed later if needed. Internally, obviously, any geometry type can be used for the `Boundable` geometry.